neurolib/utils/signal.py

```diff
@@ -0,0 +1,515 @@
+"""
+Base classes for representing signals.
+"""
+
+import logging
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
+from mne import set_log_level
+from mne.filter import filter_data, resample
```

Comment:
If you could do this without mne but with scipy.filter (which we already require) you'd be the mvp!